recipe-server/client/control/tests/reducers/test_controlAppReducer.js

```diff
@@ -207,18 +211,53 @@ describe('controlApp reducer', () => {
       ...initialState,
       recipes: {
         ...initialState.recipes,
-        list: [{
-          id: 1,
-          name: 'Lorem Ipsum',
+        list: {
+          1: {
+            id: 1,
+            name: 'Lorem Ipsum',
+            enabled: true,
+          },
+          2: {
+            id: 2,
+            name: 'Dolor set amet',
+            enabled: true,
+          },
+        },
+      },
+    });
+  });
+
+  describe('Recipe Creation and Updating', () => {
+    it('only has one recipe in the `list` after creating and updating a new recipe', async () => {
```

Comment:
Aight - took some playing but I figured out the issue. The bug didn't lie with updating the recipe, it was creating the recipe and then loading the recipe list. Creating a recipe stored it in memory, then loading the recipe list from the API would add just append to the list in memory, not checking against dupes.

I had figured it out using a UI test, but in the end a reducer test was all that was really needed, so that's the test that went in. I also confirmed the test failed appropriately on `master`, as well.

This should be all set!